src/core/qgsvectorfilewriter.cpp

```diff
@@ -2253,6 +2263,29 @@ gdal::ogr_feature_unique_ptr QgsVectorFileWriter::createFeature( const QgsFeatur
 
       case QVariant::Invalid:
         break;
+
+      case QVariant::List:
+        // only string list supported at the moment, fall through to default for other types
+        if ( field.subType() == QVariant::String )
+        {
+          QStringList list = attrValue.toStringList();
+          int count = list.count();
+          char **lst = new char *[count];
```

Comment:
who's is in charge to free lst? possible leak?

Reply:
@luipir , after looking at gdal's source code, we are in charge.

I was seeking enlightenment as to how to free a char **, that'd be my TIL moment ;)

Reply:
I'm scared to free after gdal call... no idea in what part or code have to be fee without crash. Here @rouault is necessary